### PR TITLE
refactor(zimbra): add missing loadings & cache invalidations

### PR DESCRIPTION
**Pull request by tristanwagner**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/zimbra-ga` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-15534 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/zimbra/src/hooks/useMailingList.ts [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
-  });
-
-  return {
-    isLoading,
-    isError,
-    error,
-    data,
-  };
+    enabled:
+      (typeof options.enabled !== 'undefined' ? options.enabled : true) &&
```
Comment (antonymarion) [inline packages/manager/apps/zimbra/src/hooks/useMailingList.ts:32]: === 'boolean' or anything else than boolean ( !== 'undefined' ) ?

File: packages/manager/apps/zimbra/src/hooks/useMailingLists.ts [hunk 2]
```diff
@@ -24,19 +35,15 @@ export const useMailingLists = (
...
-  });
-
-  return {
-    isLoading,
-    isError,
-    error,
-    data,
-  };
+    enabled:
+      (typeof options.enabled !== 'undefined' ? options.enabled : true) &&
```
Comment (antonymarion) [inline packages/manager/apps/zimbra/src/hooks/useMailingLists.ts:46]: same

File: packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/EmailAccountSettings.page.tsx [hunk 3]
```diff
@@ -144,107 +149,78 @@ export default function EmailAccountSettings({
...
     let dataBody = {
       email: `${account}@${domain}`,
     };
-
     Object.entries(form).forEach(([key, { value }]) => {
       if (
         ![
-          ...['account', 'domain'],
-          ...[form.password.value === '' ? 'password' : ''],
+          'account',
```
Comment (antonymarion) [inline packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/EmailAccountSettings.page.tsx:214]: do we store constants values like these in constants files instead?

File: packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx [hunk 4]
```diff
@@ -213,59 +216,59 @@ export default function MailingListSettings({
...
+      addSuccess(
+        <OsdsText
+          color={ODS_THEME_COLOR_INTENT.text}
+          size={ODS_THEME_TYPOGRAPHY_SIZE._100}
+          level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+          hue={ODS_THEME_COLOR_HUE._500}
+        >
+          {t(
+            editMailingListId
+              ? 'zimbra_mailinglist_edit_success_message'
```
Comment (antonymarion) [inline packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx:235]: same for constants here

File: packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx [hunk 5]
```diff
@@ -213,59 +216,59 @@ export default function MailingListSettings({
...
+      addError(
+        <OsdsText
+          color={ODS_THEME_COLOR_INTENT.text}
+          size={ODS_THEME_TYPOGRAPHY_SIZE._100}
+          level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+          hue={ODS_THEME_COLOR_HUE._500}
+        >
+          {t(
+            editMailingListId
+              ? 'zimbra_mailinglist_edit_error_message'
```
Comment (antonymarion) [inline packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx:252]: same for constants here

File: packages/manager/apps/zimbra/src/hooks/useMailingList.ts (hunk 1 above)
Comment (tristanwagner) [reply packages/manager/apps/zimbra/src/hooks/useMailingList.ts:32]: it just means that if `enabled` wasn't passed in the options we consider it true

File: packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/EmailAccountSettings.page.tsx (hunk 3 above)
Comment (tristanwagner) [reply packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/EmailAccountSettings.page.tsx:214]: we do not at the moment

File: packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx (hunk 4 above)
Comment (tristanwagner) [reply packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx:235]: same we don't store translations keys as constants

File: packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx (hunk 5 above)
Comment (tristanwagner) [reply packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx:252]: same we don't store translations keys as constants

**Review by ghyenne — APPROVED** [review]

**Review by antonymarion — APPROVED** [review]

**Pull request merged**